plugins/amoeba/platforms/common/src/kernels/amoebaGk.cc

```diff
@@ -3,27 +3,57 @@
 /**
  * Reduce the Born sums to compute the Born radii.
  */
-KERNEL void reduceBornSum(GLOBAL const mm_long* RESTRICT bornSum, GLOBAL const float2* RESTRICT params, GLOBAL real* RESTRICT bornRadii) {
+KERNEL void reduceBornSum(GLOBAL const mm_long* RESTRICT bornSum, GLOBAL const float4* RESTRICT params, GLOBAL real* RESTRICT bornRadii) {
+
+    real RECIP_MAX_RADIUS3 = POW(BIG_RADIUS, (real) -3.0);
+    real PI4_3 = 4.0 / 3.0 * M_PI;
+    real INVERSE_PI4_3 = 1.0 / PI4_3;
+    real ONE_THIRD = 1.0 / 3.0;
```

Comment:
Writing double precision constants in kernel code leads to the GPU performing double precision operations, even if it's just to cast the value back to single precision.  This leads to much slower code.  When possible prefer integer constants, or if not then single precision.  This takes a lot of care in how you write expressions.  I would write these lines as

```c++
    real RECIP_MAX_RADIUS3 = POW(BIG_RADIUS, (real) -3);
    real PI4_3 = 4 * M_PI / 3;
    real INVERSE_PI4_3 = 1 / PI4_3;
    real ONE_THIRD = 1 / (real) 3;
```